mir/src/syntax.rs

```diff
@@ -752,17 +752,325 @@ impl Program {
         println!("fn{f}:_{var0} = {val0:?}\n_{var1} = {val1:?}\n_{var2} = {val2:?}\n_{var3} = {val3:?}");
     }
     "#;
+    // Implements printf based debuggig for primitive types.
+    pub const PRINTF_DUMPER: &'static str = r#"
+    use std::ffi::{c_char, c_int};
 
+    extern "C" {
+        fn printf(fmt: *const c_char, ...) -> c_int;
+    }
+    trait PrintFDebug{
+        fn printf_debug(&self);
```

Comment:
Could you make it `unsafe fn` here and in the implementations instead of using unsafe blocks?